File: C#/백준/Bronze/29267. Случай с игрой/Случай с игрой.cs
using System;

namespace q{
    class w{
        static void Main(){
            int[] aa = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            int cnt = 0;
            int save = 0;
            for (int i=0; i<aa[0]; i++){
                string a = Console.ReadLine();
                if(a[1]=='a'){
                    save = cnt;
                    Console.WriteLine(cnt);
                }
                if(a[1]=='o'){
                    cnt=save;
                    Console.WriteLine(cnt);
                }
                if(a[1]=='h'){
                    Console.WriteLine(--cnt);
                }
                if(a[1]=='m'){
                    cnt+=aa[1];
                    Console.WriteLine(cnt);
                }
            }
        }
    }
}

File: C++11/백준/Bronze/29267. Случай с игрой/Случай с игрой.cc
#include <iostream>
using namespace std;

int main(){
    int a,b,save=0,cnt=0;
    string s;
    cin >> a>> b;
    for (int i=0; i<a; i++){
        cin >>s;
        if(s[1]=='a'){
            save = cnt;
            cout << cnt << endl;
        }
        if(s[1]=='o'){
            cnt = save;
            cout << cnt << endl;
        }
        if(s[1]=='h'){
            cout << --cnt << endl;
        }
        if(s[1]=='m'){
            cnt += b;
            cout << cnt <<endl;
        }
    }
}